cpp/tests/io/orc_test.cpp

```diff
@@ -1097,6 +1097,32 @@ TEST_F(OrcReaderTest, SingleInputs)
   CUDF_TEST_EXPECT_TABLES_EQUAL(*result.tbl, *table1);
 }
 
+TEST_F(OrcReaderTest, ZstdCompressionRegression)
+{
+  // Test with zstd compressed orc file with high compression ratio.
+  constexpr uint8_t input_buffer[] = {
+    0x4f, 0x52, 0x43, 0x5a, 0x00, 0x00, 0x28, 0xb5, 0x2f, 0xfd, 0xa4, 0x34, 0xc7, 0x03, 0x00, 0x74,
+    0x00, 0x00, 0x18, 0x41, 0xff, 0xaa, 0x02, 0x00, 0xbb, 0xff, 0x45, 0xc8, 0x01, 0x25, 0x30, 0x04,
+    0x65, 0x00, 0x00, 0x10, 0xaa, 0x1f, 0x02, 0x00, 0x01, 0x29, 0x0b, 0xc7, 0x39, 0xb8, 0x02, 0xcb,
+    0xaf, 0x38, 0xc0, 0x07, 0x00, 0x00, 0x40, 0x01, 0xc0, 0x05, 0x00, 0x00, 0x46, 0x4d, 0x45, 0x00,
+    0x00, 0x0a, 0x06, 0x08, 0x01, 0x10, 0x01, 0x18, 0x30, 0x0a, 0x06, 0x08, 0x02, 0x10, 0x01, 0x18,
+    0x06, 0x0a, 0x06, 0x08, 0x03, 0x10, 0x01, 0x18, 0x05, 0x12, 0x02, 0x08, 0x00, 0x12, 0x04, 0x08,
+    0x03, 0x10, 0x02, 0x59, 0x00, 0x00, 0x08, 0x03, 0x10, 0x63, 0x1a, 0x0c, 0x08, 0x03, 0x10, 0x00,
+    0x18, 0x3b, 0x20, 0x25, 0x28, 0xa0, 0x9e, 0x75, 0x22, 0x10, 0x08, 0x0c, 0x12, 0x01, 0x01, 0x1a,
+    0x09, 0x63, 0x64, 0x5f, 0x67, 0x65, 0x6e, 0x64, 0x65, 0x72, 0x22, 0x02, 0x08, 0x07, 0x30, 0xa0,
+    0x9e, 0x75, 0x08, 0x2f, 0x10, 0x05, 0x18, 0x80, 0x80, 0x10, 0x22, 0x02, 0x00, 0x0c, 0x28, 0x00,
+    0x30, 0x09, 0x82, 0xf4, 0x03, 0x03, 0x4f, 0x52, 0x43, 0x17};
+
+  auto source =
+    cudf::io::source_info(reinterpret_cast<const char*>(input_buffer), sizeof(input_buffer));
+  cudf_io::orc_reader_options in_opts =
+    cudf_io::orc_reader_options::builder(source).use_index(false);
+
+  cudf::io::table_with_metadata result;
+  CUDF_EXPECT_NO_THROW(result = cudf_io::read_orc(in_opts));
```

Comment:
I don't think this check is needed. We don't usually check this if there are any checks of the actual result.
```suggestion
  auto const result = cudf_io::read_orc(in_opts);
```

Reply:
I think this check is needed to make the failure case clear.  Attempting to read this zstd compressed file without the fix in this PR causes a cuda illegal address exception.

Reply:
I actually added the num_rows check as an additional verification to make sure it is correct, but the real test is checking to see if it throws.